tools/Generator.CLI/ConsoleShell.cs

```diff
@@ -83,22 +86,36 @@ private static void SaveContent(string file, string content, bool format)
 						.ToFullString();
 				}
 
+				if (System.IO.File.Exists(file))
+				{
+					Console.WriteLine($"Deleting old file at {file}.");
+					System.IO.File.Delete(file);
+				}
+
 				Console.WriteLine($"Writing file {file}.");
-				System.IO.File.WriteAllText(file, content, Encoding.UTF8);
-				Console.WriteLine($"success.");
+
+				using (var fileStream = new FileStream(file, FileMode.Create, FileAccess.Write, FileShare.Write, 1024, true))
+				{
+					using (var streamWriter = new StreamWriter(fileStream, Encoding.UTF8))
```

Comment:
Unfortunately, this code also writes byte order marks. Here are two options to avoid them:

- ```cs
  System.IO.File.WriteAllText(file, content);
  ```
  (https://source.dot.net/#System.IO.FileSystem/System/IO/File.cs,3768ad284a63029f)

- ```cs
  using (StreamWriter sw = new StreamWriter(file))
  {
      sw.Write(content);
  }
  ```
  (https://source.dot.net/#System.Private.CoreLib/shared/System/IO/StreamWriter.cs,f7abd08ea061985d)
  (https://source.dot.net/#System.IO.FileSystem/System/IO/File.cs,3768ad284a63029f)

Reply:
If you want to explicitly specify UTF8-no-BOM, you use `new UTF8Encoding(encoderShouldEmitUTF8Identifier: false)`. It's ugly.

Reply:
went for StreamWriter([string]path) version, since you obviously prefered that one

Reply:
I don't prefer it (I would have used WriteAllTextAsync), which is why I suggested `WriteAllText(file, content)` in my initial comment, but it also doesn't bother me at all either way.